metronome/config/specs/src/io/iohk/metronome/config/ConfigParserSpec.scala

```diff
@@ -22,6 +22,24 @@ class ConfigParserSpec
     json.noSpaces shouldBe """{"nested-structure":{"bar_baz":{"spam":"eggs"},"foo":10}}"""
   }
 
+  "toJsonEscape" should "parse and escape escape.conf to JSON" in {
+    val conf = ConfigFactory.parseString(
+      """{"java_home":"C:\\Program Files\\java\\jdk-8\""}"""
+    )
+    val orig = ConfigParser.toJson(conf.root())
+    val json = ConfigParser.withCamelCase(orig)
+
+    val env = Map(
+      "JAVA_HOME" -> """C:\Program Files\java\jdk-11""""
+    )
+
+    val result = ConfigParser.withEnvVarOverrides(json, "", env)
+
+    inside(result) { case Right(json) =>
+      json.noSpaces shouldBe """{"javaHome":"C:\\Program Files\\java\\jdk-11\""}"""
```

Comment:
It seems a bit strange when you override 
`C:\\Program Files\\java\\jdk-8\`   <- see the single backslash at the end 
with
`C:\Program Files\java\jdk-11` <- no backslash at the end
but expect 
`C:\\Program Files\\java\\jdk-11\`  <- with single backslash at the end

Or maybe I couldn't get what does the test check.
If it is testing only json escape symbols, then perhaps we can test different strings, without a config for these sequences: 
```
      char = unescaped /
          escape (
              %x22 /          ; "    quotation mark  U+0022
              %x5C /          ; \    reverse solidus U+005C
              %x2F /          ; /    solidus         U+002F
              %x62 /          ; b    backspace       U+0008
              %x66 /          ; f    form feed       U+000C
              %x6E /          ; n    line feed       U+000A
              %x72 /          ; r    carriage return U+000D
              %x74 /          ; t    tab             U+0009
              %x75 4HEXDIG )  ; uXXXX                U+XXXX

      escape = %x5C              ; \

      quotation-mark = %x22      ; "

      unescaped = %x20-21 / %x23-5B / %x5D-10FFFF
```

Reply:
```
val conf = ConfigFactory.parseString(
      """{"java_home":"C:\\Program Files\\java\\jdk-8\""}"""
                                                      ^
```

It's not intuitive but there _is_ a `"` in the original, which is escaped as `\"`, followed by the actual `"` that closes the string, so the `java_home` path has a double quote in it.

The override _also_ has a double quote in it, but this time it doesn't need escaping because of triple quoting:
```
    val env = Map(
      "JAVA_HOME" -> """C:\Program Files\java\jdk-11""""
                                                    ^
```

Reply:
Thanks for the recommendation, I applied clearer examples